packages/publisher/github/test/github_spec.ts

```diff
@@ -104,5 +104,17 @@ describe('GitHub', () => {
         gh.getGitHub();
       }).to.throw('Please set GITHUB_TOKEN in your environment to access these features');
     });
+
+    it('should synthesize name without leading or trailing periods', () => {
+      expect(GitHub.sanitizeName('path/to/.foo.'), 'foo');
+    });
+
+    it('should synthesize name without multiple periods', () => {
+      expect(GitHub.sanitizeName('path/to/foo..bar'), 'foobar');
+    });
+
+    it('should synthesize name without non-alphanumeric characters', () => {
+      expect(GitHub.sanitizeName('path/to/foo%bar baz.'), 'foo-bar-baz');
+    });
```

Comment:
suggestion(naming, organization): `sanitizeName` should be in its own `describe` block and the test names should be clearer.
```suggestion
  });

  describe('sanitizeName', () => {
    it('should remove leading and trailing periods from the basename', () => {
      expect(GitHub.sanitizeName('path/to/.foo.'), 'foo');
    });

    it('should remove multiple periods in a row', () => {
      expect(GitHub.sanitizeName('path/to/foo..bar'), 'foobar');
    });

    it('should replace non-alphanumeric, non-hyphen characters with hyphens', () => {
      expect(GitHub.sanitizeName('path/to/foo%bar baz.'), 'foo-bar-baz');
    });
```